test/test_appengine.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+"""
+Tests the ssl compatibility module for appengine.
+"""
+import hyper
+from hyper import ssl_compat_appengine
+from server import SocketLevelTest
+import socket
+import pytest
+
+
+class TestAppengine(object):
+    """
+    Test cases for ssl_compat_appengine module.
+    """
+
+    def test_field_existences(self):
+        assert ssl_compat_appengine.PROTOCOL_TLSv1 is not None
+        assert ssl_compat_appengine.PROTOCOL_SSLv23 is not None
+        assert ssl_compat_appengine.PROTOCOL_SSLv3 is not None
+        assert ssl_compat_appengine.CERT_NONE is not None
+        assert ssl_compat_appengine.CERT_OPTIONAL is not None
+        assert ssl_compat_appengine.CERT_REQUIRED is not None
+        assert ssl_compat_appengine.OP_NO_COMPRESSION is not None
+        assert ssl_compat_appengine.match_hostname is not None
+        assert ssl_compat_appengine.SSLContext is not None
+
+    def test_SSLContext(self):
+        context = ssl_compat_appengine.SSLContext(
+            ssl_compat_appengine.PROTOCOL_SSLv23)
+        context.set_default_verify_paths()
+        assert context.protocol_version == ssl_compat_appengine.PROTOCOL_SSLv23
+
+
+class TestAppengineSocket(SocketLevelTest):
+    """
+    Test case for wrap_socket.
+    """
+    h2 = False
+
+    def socket_handler(self, listener):
+        sock = listener.accept()[0]
+        sock.close()
+
+    def test_wrap_socket(self):
+        self.set_up()
+        self._start_server(self.socket_handler)
+        context = ssl_compat_appengine.SSLContext(
+            ssl_compat_appengine.PROTOCOL_SSLv23)
+        context.set_default_verify_paths()
+        context.verify_mode = ssl_compat_appengine.CERT_NONE
+        context.verify_hostname = False
+        # This invocation makes sure it does not fail.
+        context.load_verify_locations('test/certs/server.crt')
+        sock = socket.create_connection(
+            (self.server_thread.host, self.server_thread.port))
+        ssl_sock = context.wrap_socket(sock)
+        assert ssl_sock.selected_npn_protocol() == 'h2'
+        ssl_sock.close()
+        self.tear_down()
```

Comment:
GAE doesn't offer a different ssl library actually, it's the same ssl as bundled in future version of Python, but the values in the ssl module is slightly limited. SSLContext doesn't exist, for example.

That means that this ssl_compat_appengine should work with the normal ssl module outside of AppEngine (if it's newer than 2.7.9), and the test case would be still useful to verify the behavior of the module itself.

For the second comment, thank you for catching that. I put sock.do_handshake() on socket_handler so that it verifies to establish the connection, so I think that's good enough.